mp/src/game/shared/neo/neo_gamerules.cpp

```diff
@@ -1828,24 +1867,86 @@ static CNEO_Player* FetchAssists(CNEO_Player* attacker, CNEO_Player* victim)
 	return NULL;
 }
 
+#ifdef GAME_DLL
+void CNEORules::CheckIfCapPrevent(CNEO_Player *capPreventerPlayer)
+{
+	// If this is the only player alive left before the suicide/disconnect and the other team was holding
+	// the ghost, reward the other team an XP to the next rank as a ghost cap was prevented.
+	if (neo_sv_suicide_prevent_cap_punish.GetBool() && m_nRoundStatus == NeoRoundStatus::RoundLive &&
+			!m_bTeamBeenAwardedDueToCapPrevent)
+	{
+		bool bOtherTeamPlayingGhost = false;
+		int iTallyAlive[TEAM__TOTAL] = {};
+		const int iPreventerTeam = capPreventerPlayer->GetTeamNumber();
+		// Sanity check: Make sure it's only Jinrai/NSF players
+		const bool bValidTeam = iPreventerTeam == TEAM_JINRAI || iPreventerTeam == TEAM_NSF;
+		Assert(bValidTeam);
+		if (bValidTeam)
+		{
```

Comment:
Nit: could `return` here for `!bValidTeam`, so we could collapse the indentation for readability.

Likewise I'd prefer if we early-returned on the inverse of the first condition of this function, instead of everything being wrapped inside the if-scope, but maybe personal preference.